docs/website/docs/developers/debugging/sanitizers.md

```diff
@@ -39,6 +39,58 @@ Enabling ASan in the IREE build is a simple matter of setting the
 cmake -DIREE_ENABLE_ASAN=ON ...
 ```
 
+#### Linking to the dynamic Asan runtime
+
+You may want to use ASan when using the python bindings.
+To achieve this you need to link to the ASan runtime dynamically instead of
+linking it statically into an executable.
+
+```shell
+cmake \
+  -DIREE_ENABLE_ASAN=ON \
+  -DCMAKE_EXE_LINKER_FLAGS=-shared-libasan \
+  -DCMAKE_SHARED_LINKER_FLAGS=-shared-libasan \
+  -DCMAKE_C_COMPILER=clang-12 \
+  -DCMAKE_CXX_COMPILER=clang++-12 \
+  ...
+```
+
+Then when running things the ASan runtime will have to be preloaded.
+```shell
+LD_PRELOAD=/usr/lib/llvm-12/lib/clang/12.0.0/lib/linux/libclang_rt.asan-x86_64.so \
+ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-12/bin/llvm-symbolizer \
+  python ...
+```
+
+If you want to run the IREE CUDA runtime driver it is likely you would need.
+```shell
+ASAN_OPTIONS="protect_shadow_gap=0"
+```
+
+Like this
+```shell
+LD_PRELOAD=/usr/lib/llvm-12/lib/clang/12.0.0/lib/linux/libclang_rt.asan-x86_64.so \
+ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-12/bin/llvm-symbolizer \
+ASAN_OPTIONS="protect_shadow_gap=0" \
+  python ...
+```
+
+On Ubuntu the corresponding ASan runtime is provided by a package like
+`libclang-common-12-dev` depending on your Clang version.
+E.g.
+```shell
+sudo apt install libclang-common-12-dev llvm-12 clang-12
+```
+
+Note that during building you would still need to preload the ASan runtime, since
+the build executes its own binaries that are linked against the runtime.
+```shell
+LD_PRELOAD=/usr/lib/llvm-12/lib/clang/12.0.0/lib/linux/libclang_rt.asan-x86_64.so \
+ASAN_OPTIONS=detect_leaks=0 \
+ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-12/bin/llvm-symbolizer \
+  cmake --build ...
+```
```

Comment:
Which part of the build is this? Tools, samples, tests, etc.? If just tests/samples, it might be easier to suggest people turn those off.

Reply:
ASan affects the whole build. See [iree_setup_toolchain.cmake](https://github.com/openxla/iree/blob/df784ceabab1e7bc428f0340f1036c44a48669f0/build_tools/cmake/iree_setup_toolchain.cmake#L104).